app/src/main/java/com/infomaniak/mail/utils/Extensions.kt

```diff
@@ -305,3 +312,43 @@ fun Fragment.createDescriptionDialog(
         .setNegativeButton(R.string.buttonCancel, null)
         .create()
 }
+
+@SuppressLint("ResourceAsColor")
+fun Fragment.createInputDialog(
+    @StringRes title: Int,
+    @StringRes hint: Int,
+    @StringRes confirmButtonText: Int,
+    onPositiveButtonClicked: (Editable?) -> Unit,
+) = with(DialogInputBinding.inflate(layoutInflater)) {
+
+    fun Button.setButtonEnablement(isInputEmpty: Boolean) {
+        isEnabled = !isInputEmpty
+
+        val (textColor, backgroundColor) = if (isInputEmpty) {
+            R.color.disabledDialogButtonTextColor to resources.getColor(R.color.backgroundDisabledDialogButton, null)
```

Comment:
```suggestion
fun Fragment.createInputDialog(
    @StringRes title: Int,
    @StringRes hint: Int,
    @StringRes confirmButtonText: Int,
    onPositiveButtonClicked: (Editable?) -> Unit,
) = with(DialogInputBinding.inflate(layoutInflater)) {

    fun Button.setButtonEnablement(isInputEmpty: Boolean) {
        isEnabled = !isInputEmpty

        val (textColor, backgroundColor) = if (isInputEmpty) {
            @Suppress("ResourceAsColor")
            R.color.disabledDialogButtonTextColor to resources.getColor(R.color.backgroundDisabledDialogButton, null)
```

In case the lint is actually useful somewhere else inside the function, I like to put the suppress right next to the faulty line rather than disable it for the whole function